client/sync.go

```diff
@@ -0,0 +1,129 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/topfreegames/eventsgateway/metrics"
+	pb "github.com/topfreegames/protos/eventsgateway/grpc/generated"
+	"google.golang.org/grpc"
+)
+
+type gRPCClientSync struct {
+	client pb.GRPCForwarderClient
+	config *viper.Viper
+	conn   *grpc.ClientConn
+	logger logrus.FieldLogger
+}
+
+func newGRPCClientSync(
+	configPrefix string,
+	config *viper.Viper,
+	logger logrus.FieldLogger,
+	serverAddress string,
+	client pb.GRPCForwarderClient,
+	opts ...grpc.DialOption,
+) (*gRPCClientSync, error) {
+	g := &gRPCClientSync{
+		config: config,
+		logger: logger,
+	}
+	if err := g.configureGRPCForwarderClient(
```

Comment:
nit: remove line breaks? This is fine for Go.

Reply:
ditto